nixos/modules/services/x11/xserver.nix

```diff
@@ -651,22 +651,31 @@ in
     system.extraDependencies = singleton (pkgs.runCommand "xkb-layouts-exist" {
       inherit (cfg) layout xkbDir;
     } ''
-      if sed -n -e ':i /^! \(layout\|variant\) *$/ {
-          :l; n; /^!/bi; s/^ *\([^ ]\+\).*/\1/p; tl
-         }' "$xkbDir/rules/base.lst" | grep -qxF "$layout"
-      then
-        touch "$out"
+      IFS=, read -a layouts <<<'${cfg.layout}'
```

Comment:
This is not well escaped, because if the `layout` option contains a single quote you can inject arbitrary shell commands, like let's say `layout = "' || exit 0`. In this case it's better to use `$layout` here, because we already have that in the builder environment.